content/en/blog/_posts/2023-05-05-memory-qos-cgroups-v2/index.md

```diff
@@ -0,0 +1,286 @@
+$memory.min = \sum_{i=0}^{no. of pods}pod.spec.containers[i].resources.requests[memory]$  
+
+`memory.min` in node level cgroup:  
+$memory.min = \sum_{i}^{no. of nodes}\sum_{j}^{no. of pods}pod[i].spec.containers[j].resources.requests[memory]$  
+
+Kubelet will manage the cgroups hierarchy of the pod level and node level cgroups 
+directly using the libcontainer library (from the runc project), while container
+cgroups limits are managed by the container runtime.
+
+### Support for Pod QoS classes
+
+Based on user feedback for the Alpha feature in Kubernetes v1.22, some users would like 
+to opt out of MemoryQoS on a per-pod basis to ensure there is no early memory throttling.
+Therefore, in Kubernetes v1.27 Memory QOS also supports memory.high to be set as per 
+Quality of Service(QoS) for Pod classes. Following are the different cases for memory.high
+as per QOS classes:
+
+1. **Guaranteed pods** by their QoS definition require memory requests=memory limits and are 
+not overcommitted. Hence MemoryQoS feature is disabled on those pods by not setting 
+memory.high. This ensures that Guaranteed pods can fully use their memory requests up
+to their set limit, and not hit any throttling.
+
+2. **Burstable pods** by their QoS definition require at least one container in the Pod with
+CPU or memory request or limit set.
+    
+   * When requests.memory and limits.memory are set, the formula is used as-is:
+
+     {{< figure src="/blog/2023/05/05/qos-memory-resources/container-memory-high-limit.svg" title="`memory.high` when requests and limits are set" alt="`memory.high` when requests and limits are set" >}}
+
+   * When requests.memory is set, limits.memory is not set, we substitute limits.memory
+     for node allocatable memory in the formula:
+
+     {{< figure src="/blog/2023/05/05/qos-memory-resources/container-memory-high-no-limits.svg" title="`memory.high` when requests and limits are not set" alt="`memory.high` when requests and limits are not set" >}}
+
+3. **BestEffort** by their QoS definition do not require any memory or CPU limits or requests.
+   For this case, kubernetes sets requests.memory = 0 and substitute limits.memory for node allocatable
+   memory in the formula:
+
+   {{< figure src="/blog/2023/05/05/qos-memory-resources/container-memory-high-best-effort.svg" title="`memory.high` for BestEffort Pod" alt="`memory.high` for BestEffort Pod" >}}
+
+**Summary**: Only Pods in Burstable and BestEffort QoS classes will set `memory.high`.
+Guaranteed QoS pods do not set `memory.high` as their memory is guaranteed.
+
+## How do I use it?
+
+The prerequisites for enabling Memory QoS feature on your Linux node are:
+
+1. Verify the [requirements](/docs/concepts/architecture/cgroups/#requirements) 
+   related to [Kubernetes support for cgroups v2](/docs/concepts/architecture/cgroups)
+   are met. 
+2. Ensure CRI Runtime supports Memory QoS. In Alpha phase of this feature,
+   only containerd and cri-o are supported. Related PRs:
```

Comment:
```suggestion
1. Ensure CRI Runtime supports Memory QoS. At the time of writing, only
   containerd and CRI-O provide support compatible with this alpha. That was
   implemented in the following PRs:
```

Kubernetes doesn't provide support for runtimes' features; mostly I see it as CRI implementations providing support for Kubernetes features. The CRI is defined by _this_ project, and runtime adapters can choose to conform to it.